summarization/stamp_page_picking/scoring_utils.py

```diff
@@ -0,0 +1,146 @@
+''' Module for scoring utils '''
+
+from sklearn.metrics.pairwise import cosine_similarity
+
+
+class ScoringUtils:
+    def __init__(self, stamp_pages, stamp_page_covers, cover_size):
+        self.stamp_pages = stamp_pages
+        self.stamp_page_covers = stamp_page_covers
+        self.cover_size = cover_size
+        self.picked_cover = [0] * self.cover_size
+
+        # sweeping index is initially the maximum or last index possible
+        self.sweeping_index = max(
+            [max(page.media_index, page.sentence_index)
+             for page in self.stamp_pages]
+        )
+
+        self.last_picked_stamp_page_index = -1
+
+        # step size to change sweeping index
+        # TODO: add variable change in step size
+        # based on iteration
+        self.step_size = 1
+        self.iteration_count = 1
+
+        self.individual_score_weights = [
+            1,  # content change score weight
+            1,  # sentiment change score
+            1,  # textual entailment score
+            1,  # unpicked weights/cost score - as used in max cover
+                # it tells how many sentences in the web-page are
+                # unpicked yet and its ratio to stamp page cost
+            1   # sweeping index score
+        ]
+
+    def _update_last_picked_stamp_page(self, stamp_index):
+        self.last_picked_stamp_page_index = stamp_index
+
+    def _interestingess_metric(self, stamp_page_index):
+        '''
+        calculates the interesting-ness metric based
+        on the last picked stamp page
+        '''
+        from_stamp_page_index = self.last_picked_stamp_page_index
+        to_stamp_page_index = stamp_page_index
+
+        content_change_score \
+            = self._get_content_change_score(
+                from_stamp_page_index,
+                to_stamp_page_index
+            )
+
+        sentiment_change_score \
+            = 0  # will be amended once sentiment detection PR is added
+
+        content_similarity_score = self._get_content_similarity_score(
+            from_stamp_page_index, to_stamp_page_index)
+
+        unpicked_weight_cost_ratio_score \
+            = self._get_unpicked_weights_to_cost_ratio_score(stamp_page_index)
+
+        sweeping_index_score = self._get_distance_from_sweeping_index(
+            stamp_page_index)
+
+        return self._compute_weighted_score(
```

Comment:
Been running this locally and I think we should add a score on is_embedded_content - The pages I'm getting now have lots of text (will try out on a few more pages).

I think we'll be fine if it heavily favors embedded content, and if there's a mix we should get that. Something like if we have 5 images, 10 tweets, 20 text stamp pages - If we have 5 pages to fill - 2 images, 2 tweets, and one/zero (pick another tweet) text should be good.

Reply:
That's right it is very text heavy right now - this shouldn't be the case - trying to find why this happens . 
This should probably be solved when the weights are tweaked properly.